src/probnum/diffeq/steprule.py

```diff
@@ -19,66 +24,109 @@ def is_accepted(self, proposedstep, errorest, **kwargs):
-    Warning is printed if the suggested step is smaller than roughly
-    machine precision: 1e-15.
+# Once we have other controls, e.g. PI control, we can rename this into ProportionalControl.
+# Until then, lets keep the delta small, I'd say (N).
+class AdaptiveSteps(StepRule):
+    """Adaptive step size selection using proportional control.
 
     Parameters
     ----------
-    tol_per_step : float
-        Tolerance per step (absolute)
-    localconvrate : float
-        (Estimated) convergence rate of the solver.
+    firststep : float
+        First step to be taken by the ODE solver (which happens in absence of error estimates).
     limitchange : list with 2 elements, optional
         Lower and upper bounds for computed change of step.
     safetyscale : float, optional
         Safety factor for proposal of distributions, 0 << safetyscale < 1
+    minstep : float, optional
+        Minimum step that is allowed. A runtime error is thrown if the proposed step is smaller. Default is 1e-15.
+    maxstep : float, optional
+        Maximum step that is allowed. A runtime error is thrown if the proposed step is larger. Default is 1e15.
     """
 
     def __init__(
         self,
-        tol_per_step,
-        localconvrate,
+        firststep,
         limitchange=(0.1, 5.0),
         safetyscale=0.95,
-        **kwargs
+        minstep=1e-15,
+        maxstep=1e15,
     ):
-        self.tol_per_step = float(tol_per_step)
         self.safetyscale = float(safetyscale)
-        self.localconvrate = float(localconvrate + 1)
         self.limitchange = limitchange
+        self.minstep = minstep
+        self.maxstep = maxstep
+
+        super().__init__(firststep=firststep)
 
-    def suggest(self, laststep, errorest, **kwargs):
+    def suggest(self, laststep, scaled_error, localconvrate=None):
         small, large = self.limitchange
-        ratio = self.tol_per_step / (laststep * errorest)
-        change = self.safetyscale * ratio ** (1.0 / self.localconvrate)
+
+        ratio = 1.0 / scaled_error
+        change = self.safetyscale * ratio ** (1.0 / localconvrate)
+
+        # The below code should be doable in a single line?
         if change < small:
             step = small * laststep
         elif large < change:
             step = large * laststep
         else:
             step = change * laststep
-        if step < 1e-15:
-            print("Warning: Stepsize is num. zero (h=%.1e)" % step)
+
+        if step < self.minstep:
+            raise RuntimeError("Step-size smaller than minimum step-size")
+        if step > self.maxstep:
+            raise RuntimeError("Step-size larger than maximum step-size")
         return step
 
-    def is_accepted(self, proposedstep, errorest, **kwargs):
-        return errorest * proposedstep < self.tol_per_step
+    # Looks unnecessary, though maybe we want tolerance per unit step
+    # in which case it is good to have such method in here.
```

Comment:
This. Also, for constant steps we want to accept every single time, without computing anything, so I find it convenient to have this method here.

Reply:
I'm sorry but I don't understand your comment :/

Reply:
Ah I see! I believe that should be handled differently though, with E <- E / h, since the error per step or per unit should also be the one that's used for step size suggestions. Does that make sense?

Reply:
true. Didnt change the code here either way, though, and the comment is removed.